scripts/BBoxManage.sh

```diff
@@ -0,0 +1,474 @@
+#!/bin/bash
+# ------------------------------------------------------------------
+# [Kelly Davis] ssid.sh
+#               Script for configuring the ssid in hostapd
+# ------------------------------------------------------------------
+
+VERSION=0.1.0
+SUBJECT=biblebox_control_ssid_script
+USAGE="Usage: BBoxManage.sh -dhv [get|set] [ssid|channel|hostname] <value>"
+NGINX_CONFIG="/etc/nginx/sites-enabled/vhosts.conf"
+HOSTAPD_CONFIG="/etc/hostapd/hostapd.conf"
+PASSWORD_CONFIG="/usr/local/biblebox/etc/basicauth"
+PASSWORD_SALT="BBOXFOO2016"
+DEBUG=0
+SUCCESS="SUCCESS"
+FAILURE="FAILURE"
+
+# --- Options processing -------------------------------------------
+
+while getopts ":dvhg" optname
+  do
+    case "$optname" in
+      "d")
+        DEBUG=1
+        ;;
+      "v")
+        echo "Version $VERSION"
+        exit 0;
+        ;;
+      "h")
+        echo $USAGE
+        exit 0;
+        ;;
+      "?")
+        echo "Unknown option $OPTARG"
+        exit 0;
+        ;;
+      ":")
+        echo "No argument value for option $OPTARG"
+        exit 0;
+        ;;
+      *)
+        echo "Unknown error while processing options"
+        exit 0;
+        ;;
+    esac
+  done
+
+shift $(($OPTIND - 1))
+
+action=$1
+module=$2
+val=$3
+
+# --- Locks -------------------------------------------------------
+LOCK_FILE=/tmp/$SUBJECT.lock
+if [ -f "$LOCK_FILE" ]; then
+   echo "Script is already running"
+   exit
+fi
+
+trap "rm -f $LOCK_FILE" EXIT
+touch $LOCK_FILE
+
+# --- Body --------------------------------------------------------
+
+function usage () {
+  echo $USAGE
+  exit 1;
+}
+
+function backup_nginx_config () {
+  config_file=".$(basename $NGINX_CONFIG)"
+  config_path="${NGINX_CONFIG:0:${#NGINX_CONFIG}-${#config_file}}"
+
+  # Backup the original configuration file
+  if [ ! -e "${config_path}/${config_file}" ]; then
+    if [ $DEBUG == 1 ]; then
+      echo 'Backing up $NGINX_CONFIG to ${config_path}/${config_file}'
+    fi
+
+    cp $NGINX_CONFIG "${config_path}/${config_file}" 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+  fi
+}
+
+function restore_nginx_config () {
```

Comment:
Actually, I misread the code in your backup and restore functions - I thought it was backing up the most recent version of the config file, rather than performing a once-off backup the first time it's run. Sorry about that.
I think we're doing the right thing, now that I understand it - the reset should take it back to the original state created by the ansible scripts, so the general backup/restore logic is good.